wallet/types/src/keys.rs

```diff
@@ -130,6 +132,60 @@ pub struct RootKeys {
     pub root_vrf_key: ExtendedVRFPrivateKey,
 }
 
+/// Just an empty struct used as key for the DB table
+/// It only represents a single value as there can be only one root key
+#[derive(PartialEq, Eq, PartialOrd, Ord, Encode, Decode)]
+pub struct SeedPhraseConstant;
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct SeedPhrase {
+    pub mnemonic: zeroize::Zeroizing<bip39::Mnemonic>,
+}
+
+impl Encode for SeedPhrase {
+    fn encode(&self) -> Vec<u8> {
+        let mut buf = BufWriter::new(Vec::new());
+        self.encode_to(&mut buf);
+        buf.into_inner().expect("Flushing should never fail")
+    }
+
+    fn size_hint(&self) -> usize {
+        // Preallocate enough space for the longest possible word list
+        33
+    }
+
+    fn using_encoded<R, F: FnOnce(&[u8]) -> R>(&self, f: F) -> R {
+        f(self.encode().as_slice())
+    }
+
+    fn encoded_size(&self) -> usize {
+        self.encode().len()
+    }
+
+    fn encode_to<T: serialization::Output + ?Sized>(&self, dest: &mut T) {
```

Comment:
You don't really need to manually implement all these functions. Just implementing `encode_to` is more than enough. Most of these call each other. See the Encode trait and how it already has default impls for these functions.

Reply:
removed